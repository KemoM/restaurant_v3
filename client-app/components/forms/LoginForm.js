import React from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';  
import { Link } from 'react-router-dom';
import {
  Row,
  Col,
  Button,
  Card,
  CardHeader,
  CardFooter,
  CardBody,
  Form,
} from 'reactstrap';

import { loginValidation } from './validators/loginValidation';
import { renderInputField } from './renderInputField';

class LoginForm extends React.Component {
  renderAlert() {
    if (this.props.errorMessage) {
      return (<div><span><strong>Error!</strong> {this.props.errorMessage}</span></div>);
    }
  }

  render() {
    const { fields: {username, password}, loggingIn, pristine, handleSubmit } = this.props;
    return (
      <div className="flex-row align-items-center">
        <Row>
          <Col md="6">
            <Card>
              <CardHeader>
                <strong>Login</strong> Form
              </CardHeader>
              <CardBody>
                <Form onSubmit={handleSubmit} >
                  <Field type="text" id="username" name="username" label="Username" labelClass="pr-1" component={renderInputField} value={username} />
                  <Field type="password" id="password" name="password" label="Password" component={renderInputField} value={password} />
                  {this.renderAlert()}
                  <Button type="submit" disabled={pristine || loggingIn} size="sm" color="primary"><i className="fa fa-dot-circle-o" /> Login</Button>
                  {loggingIn && 
                    <img alt="" src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA==" />
                  }
                </Form>
              </CardBody>
              <CardFooter>
                <span>Or you want to register: </span>
                <Link to="/register" className="btn btn-link">Register</Link>
              </CardFooter>
            </Card>
          </Col>
        </Row>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const loggingIn = state.authentication.get('loggingIn');
  const errorMessage = state.authentication.get('error');
  return {
    loggingIn,
    errorMessage,
  };
}

const formConfiguration = {  
  form: 'loginform',
  fields: ['username', 'password'],
  validate: loginValidation,
};

export default connect(mapStateToProps)(reduxForm(formConfiguration)(LoginForm));