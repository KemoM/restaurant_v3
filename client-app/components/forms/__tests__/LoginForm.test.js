import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import {LoginForm} from '../LoginForm';

describe('LoginForm Component', () => {
  let component, props;;
 
  beforeEach(() => {
    props = {
      fields: {
        username: "username test",
        password: "password test"
      },
      loggingIn: false, 
      pristine: false, 
      handleSubmit: jest.fn()
    };
    
    component = shallow(<LoginForm  {...props} />);
  });


  it('should render without throwing an error', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

});