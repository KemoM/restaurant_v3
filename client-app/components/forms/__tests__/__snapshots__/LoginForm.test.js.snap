// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should render without throwing an error 1`] = `
<div
  className="flex-row align-items-center"
>
  <Row
    tag="div"
  >
    <Col
      md="6"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Card
        tag="div"
      >
        <CardHeader
          tag="div"
        >
          <strong>
            Login
          </strong>
           Form
        </CardHeader>
        <CardBody
          tag="div"
        >
          <Form
            onSubmit={[Function]}
            tag="form"
          >
            <Field
              component={[Function]}
              id="username"
              label="Username"
              labelClass="pr-1"
              name="username"
              type="text"
              value="username test"
            />
            <Field
              component={[Function]}
              id="password"
              label="Password"
              name="password"
              type="password"
              value="password test"
            />
            <Button
              color="primary"
              disabled={false}
              size="sm"
              tag="button"
              type="submit"
            >
              <i
                className="fa fa-dot-circle-o"
              />
               Login
            </Button>
          </Form>
        </CardBody>
        <CardFooter
          tag="div"
        >
          <span>
            Or you want to register: 
          </span>
          <Link
            className="btn btn-link"
            replace={false}
            to="/register"
          >
            Register
          </Link>
        </CardFooter>
      </Card>
    </Col>
  </Row>
</div>
`;
